cmd/release-controller/bugzilla.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/golang/glog"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+// syncBugzilla checks whether fixed bugs in a release had their
+// PR reviewed and approved by the QA contact for the bug
+func (c *Controller) syncBugzilla(key queueKey) error {
+	defer func() {
+		err := recover()
+		panic(err)
+	}()
+
+	release, err := c.loadReleaseForSync(key.namespace, key.name)
+	if err != nil || release == nil {
+		return err
+	}
+
+	glog.V(4).Infof("Verifying fixed bugs %s", release.Config.Name)
+
+	// check if verifyBugs publish step is enabled for release
+	verifyBugs, ok := release.Config.Publish[publishTypeBugzilla]
+	if !ok || verifyBugs.Disabled {
+		return nil
+	}
+
+	// get accepted tags
+	acceptedTags := sortedRawReleaseTags(release, releasePhaseAccepted)
+	if len(acceptedTags) == 0 {
+		return nil
+	}
+	var errs []error
+	// get latest accepted tag
+	tag := acceptedTags[0]
```

Comment:
So in a disaster recovery scenario you're going to verify the tags in a newer release than they really went in.  Are you sure you don't want to scan the list to find the last one you verified (if any) and verify one release at a time (i.e. find oldest N, then diff N and N+1, and then next time around scan the next, etc)?  And if you find no N, start at the oldest and walk forward?  Knowing exactly what release it went into is probably valuable.

It would be good to restructure the "find both versions" into a function that takes a release stream as input and returns the two and then add a unit test for those cases.

Reply:
I'm not sure why we need a function to "find both versions". Shouldn't it be fine to just loop through `acceptedTags` until we find the last tag that we verified?